session.cpp

```diff
@@ -65,13 +65,15 @@ Session::~Session() {
 // callback for device attach events
 void Session::attached(libusb_device *device) {
 	shared_ptr<Device> dev = probe_device(device);
-	m_available_devices.push_back(dev);
-	cerr << "Session::attached ser: " << dev->serial() << endl;
 	if (dev) {
-		if (this->m_hotplug_attach_callback) {
-			cerr << dev << endl;
-			this->m_hotplug_attach_callback(&*dev);
-			cerr << dev << endl;
+		m_available_devices.push_back(dev);
+		cerr << "Session::attached ser: " << dev->serial() << endl;
+		if (dev) {
```

Comment:
Why `if (dev)` inside of an `if (dev)`? The two lines should just move inside of the existing `if` block.